typings/artifacts.d.ts

```diff
@@ -254,16 +317,32 @@ declare global {
         pessimisticGraph: Gatherer.Simulation.GraphNode;
       }
 
+      export interface Speedline {
+        beginning: number;
+        complete: number;
+        duration: number;
+        end: number;
+        first: number;
+        speedIndex: number;
+        frames: {
+          getProgress(): number;
+          getTimeStamp(): number;
+          isProgressInterpolated(): number;
+        }[]
+      }
+
+      // TODO(bckenny): all but navigationStart could actually be undefined.
```

Comment:
these should be updated in a follow up, but it had too many places where we're treating them as always defined. Whether we were correct or not (and we'll get to fix some bugs) remains to be seen.